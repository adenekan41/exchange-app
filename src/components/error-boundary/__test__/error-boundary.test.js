/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';

import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

/* --------------------------- Internal Dependency -------------------------- */
import ErrorBoundary from '..';

describe('ErrorBoundary Component', () => {
	afterEach(cleanup);

	it('ErrorBoundary Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<ErrorBoundary>Empty</ErrorBoundary>, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	it(`shows the fallback when there's an error`, () => {
		const Throws = () => {
			return <>Heloos </>;
		};
		const { unmount } = render(
			<ErrorBoundary>
				<Throws />
			</ErrorBoundary>
		);
		unmount();
	});

	it('ErrorBoundary matches snapshot', () => {
		const tree = renderer.create(<ErrorBoundary>Empty</ErrorBoundary>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
