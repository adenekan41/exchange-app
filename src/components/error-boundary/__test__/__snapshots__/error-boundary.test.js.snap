// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component ErrorBoundary matches snapshot 1`] = `"Empty"`;
